experimenter/experimenter/nimbus_ui_new/filtersets.py

```diff
@@ -258,6 +262,33 @@ class Meta:
         ]
 
     def filter_sort(self, queryset, name, value):
+        if value in [SortChoices.END_DATE_UP, SortChoices.END_DATE_DOWN]:
+            reverse = value == SortChoices.END_DATE_DOWN
+            if value == SortChoices.END_DATE_UP:
+                default_date = datetime.max.date()
+            else:
+                default_date = datetime.min.date()
+
+            queryset_sorted = sorted(
+                queryset,
+                key=lambda exp: exp.computed_end_date
+                if exp.computed_end_date
+                else default_date,
+                reverse=reverse,
+            )
+
+            preserved_order = Case(
+                *[
+                    When(pk=exp.id, then=Value(pos))
+                    for pos, exp in enumerate(queryset_sorted)
+                ],
+                output_field=IntegerField()
+            )
```

Comment:
Huh!  That's really interesting!  I've always wondered how to do that, I've never seen this pattern before, but it makes perfect sense.  Great job!